python/semantic_kernel/connectors/memory/sql_server.py

```diff
@@ -0,0 +1,1028 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+import json
+import logging
+import re
+import struct
+import sys
+from collections.abc import AsyncIterable, Sequence
+from contextlib import contextmanager
+from io import StringIO
+from itertools import chain
+from typing import TYPE_CHECKING, Any, ClassVar, Final, Generic, TypeVar
+
+from azure.identity.aio import DefaultAzureCredential
+from pydantic import SecretStr, ValidationError, field_validator
+
+from semantic_kernel.data.const import DISTANCE_FUNCTION_DIRECTION_HELPER, DistanceFunction, IndexKind
+from semantic_kernel.data.filter_clauses.any_tags_equal_to_filter_clause import AnyTagsEqualTo
+from semantic_kernel.data.filter_clauses.equal_to_filter_clause import EqualTo
+from semantic_kernel.data.kernel_search_results import KernelSearchResults
+from semantic_kernel.data.record_definition.vector_store_model_definition import VectorStoreRecordDefinition
+from semantic_kernel.data.record_definition.vector_store_record_fields import (
+    VectorStoreRecordDataField,
+    VectorStoreRecordKeyField,
+    VectorStoreRecordVectorField,
+)
+from semantic_kernel.data.vector_search.vector_search import VectorSearchBase
+from semantic_kernel.data.vector_search.vector_search_filter import VectorSearchFilter
+from semantic_kernel.data.vector_search.vector_search_options import VectorSearchOptions
+from semantic_kernel.data.vector_search.vector_search_result import VectorSearchResult
+from semantic_kernel.data.vector_search.vectorized_search import VectorizedSearchMixin
+from semantic_kernel.data.vector_storage.vector_store import VectorStore
+from semantic_kernel.data.vector_storage.vector_store_record_collection import VectorStoreRecordCollection
+from semantic_kernel.exceptions import VectorStoreOperationException
+from semantic_kernel.exceptions.vector_store_exceptions import VectorSearchExecutionException
+from semantic_kernel.kernel_pydantic import KernelBaseSettings
+from semantic_kernel.kernel_types import OneOrMany
+from semantic_kernel.utils.feature_stage_decorator import experimental
+
+if sys.version_info >= (3, 12):
+    from typing import override  # pragma: no cover
+else:
+    from typing_extensions import override  # pragma: no cover
+if sys.version_info >= (3, 11):
+    from typing import Self  # pragma: no cover
+else:
+    from typing_extensions import Self  # pragma: no cover
+
+if TYPE_CHECKING:
+    from pyodbc import Connection, ProgrammingError
+
+
+logger = logging.getLogger(__name__)
+
+TKey = TypeVar("TKey", str, int)
+TModel = TypeVar("TModel")
```

Comment:
I see these `TKey` and `TModel` defined in a lot of different classes. Why can't they be defined in one place and imported? Most follow the same format and binding:

```python
TKey = TypeVar("TKey", bound="str")
TModel = TypeVar("TModel")
```

Reply:
the TypeVar's they behave weirdly when imported from other places, and the TKey might also be different for each connector